tests/components/cloud/test_google_config.py

```diff
@@ -341,12 +341,12 @@ async def test_sync_google_on_home_assistant_start(
     config = CloudGoogleConfig(
         hass, GACTIONS_SCHEMA({}), "mock-user-id", cloud_prefs, hass.data["cloud"]
     )
-    hass.set_state(CoreState.starting)
+    hass.set_state(CoreState.not_running)
     with patch.object(config, "async_sync_entities_all") as mock_sync:
         await config.async_initialize()
         assert len(mock_sync.mock_calls) == 0
 
-        hass.bus.async_fire(EVENT_HOMEASSISTANT_STARTED)
+        hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
```

Comment:
The test only worked before because of the event loop delay. It didn't matter what event was fired because it would always be 1 after block_till_done

Reply:
Currently `async_at_start` will fire right away if `async_start` has been called which effectively means it will always fire right away since its not going to be called when HA isn't running

I think this should have been `return hass.state is CoreState.running` https://github.com/home-assistant/core/blob/e91a38eede5131aebe1ed6d03ce9debe1994ab6a/homeassistant/helpers/start.py#L66

I'm going to merge this one and open a PR to fix `async_at_start` to wait for the `start` event or if its already been fired as I think thats how it was intended to work.  If I'm wrong about that we can close that PR

Reply:
https://github.com/home-assistant/core/pull/112878